src/main/java/io/aiven/kafka/connect/common/grouper/ParquetTopicPartitionRecordGrouper.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2021 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.grouper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.errors.SchemaProjectorException;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import io.aiven.kafka.connect.common.config.TimestampSource;
+import io.aiven.kafka.connect.common.templating.Template;
+
+class ParquetTopicPartitionRecordGrouper extends TopicPartitionRecordGrouper {
+
+    private final SchemaBasedRotator schemaBasedRotator = new SchemaBasedRotator();
```

Comment:
Composition might be preferable over inheritance

Reply:
I totally agree with you, but so far I would like to use this solution and prepare a small PR with refactoring not only for this class but for `OutputStreamWriter`, since with this release `GCS` and `S3` won't use internal common classes.